nemo_curator/modules/config.py

```diff
@@ -178,3 +179,75 @@ def __post_init__(self):
             raise ValueError(
                 f"Epsilon to extract {self.eps_to_extract} must be in eps_thresholds {self.eps_thresholds}"
             )
+
+
+@dataclass
+class RetrieverEvalSDGConfig(BaseConfig):
+    """
+    Configuration for SDG pipeline for Retriever Evals
+
+    Attributes:
+
+    """
+
+    base_url: str
+    api_key: str
+    generator_model: str = "mistralai/mixtral-8x22b-instruct-v0.1"
+    generator_url: Optional[str] = None
+    generator_api_key: Optional[str] = None
+    temperature: Optional[float] = 0.5
+    top_p: Optional[float] = 1.0
+    num_questions: Optional[int] = 1
+    max_tokens: Optional[int] = 2048
+    squad_format: Optional[bool] = False
+
+    generator_system_prompt: Optional[
+        str
+    ] = """You are data annotator, your task is
+    to generate a question for the given document. Also generate answer to the generated
+    question."""
+
+    generator_user_prompt_template: Optional[
+        str
+    ] = """Generate {n_openlines} questions and corresponding answers based on Input Document.
+    Input Document:
+    {document}
+    """
+
+    # easiness filter parameters
+    easiness_filter: str = "nvidia/nv-embedqa-e5-v5"
```

Comment:
The generator, easiness filter, and answerability filter are too tightly coupled with this config. A user shouldn't have to worry about the answerability parameters if they want to use the easiness filter. Furthermore, what if the user wants to use a different API URL for answerability compared to easiness? Under this config, they can't. I'd recommend splitting the generator, easiness, and answerability configs into separate configurations. Actually, looking at how small the resulting configs would be, it might just be worth having them as just parameters in their respective class constructors.

Reply:
The user can leave out filters from config, and it would work based on the filters mentioned in the config file. Hence, a central location for all the configs would work the best. Also, passing the parameters in the command line would make the run command quite long.

Reply:
WIP

Reply:
fixed, done